py/datastore_service.py

```diff
@@ -42,14 +43,32 @@ def serve_forever(self):
   def make_thread_locals(self):
     #try to connect forever...
     credentials = (os.environ['POSTGRES_DB'], os.environ['POSTGRES_USER'], os.environ['POSTGRES_HOST'], os.environ['POSTGRES_PASSWORD'])
-    while True:
-      try:
-        self.sql_conn = psycopg2.connect("dbname='%s' user='%s' host='%s' password='%s'" % credentials)
-        break
-      except Exception as e:
-        sys.stderr.write('Failed to connect to database with: %s' % repr(e))
-        sys.stderr.flush()
-        time.sleep(5)
+    try:
+      self.sql_conn = psycopg2.connect("dbname='%s' user='%s' host='%s' password='%s'" % credentials)
+    except Exception as e:
+      sys.stderr.write('Failed to connect to database with: %s' % repr(e))
+      sys.stdout.flush()
```

Comment:
that should be `stderr` and i would add maybe also a `sys.exit(1)` here to bomb the whole program since its not a recoverable error